src/components/image/index.tsx

```diff
@@ -1,75 +1,80 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {PureComponent} from 'react';
+//@ts-ignore
 import hoistNonReactStatic from 'hoist-non-react-statics';
-import {Image as RNImage, StyleSheet, ImageBackground} from 'react-native';
-import {Constants} from '../../helpers';
-import {PureBaseComponent} from '../../commons';
+import {Image as RNImage, ImageProps as RNImageProps, StyleSheet, ImageBackground} from 'react-native';
+import Constants from '../../helpers/Constants';
+import {asBaseComponent, forwardRef, ForwardRefInjectedProps} from '../../commons/new';
+// @ts-ignore
 import Assets from '../../assets';
-import Overlay from '../overlay';
+import Overlay, {OverlayTypeType} from '../overlay';
+
+type ImageProps = RNImageProps & {
+  /**
+   * custom source transform handler for manipulating the image source (great for size control)
+   */
+  sourceTransformer?: Function;
+  /**
+   * if provided image source will be driven from asset name
+   */
+  assetName?: string;
+  /**
+   * the asset group, default is "icons"
+   */
+  assetGroup?: string;
+  /**
+   * the asset tint
+   */
+  tintColor?: string;
+  /**
+   * whether the image should flip horizontally on RTL locals
+   */
+  supportRTL?: boolean;
+  /**
+   * Show image as a cover, full width, image (according to aspect ratio, default: 16:8)
+   */
+  cover?: boolean;
+  /**
+   * The aspect ratio for the image
+   */
+  aspectRatio?: number;
+  /**
+   * The type of overly to place on top of the image. Note: the image MUST have proper size, see examples in:
+   * https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/OverlaysScreen.js
+   */
+  overlayType?: OverlayTypeType;
+  /**
+   * Pass a custom color for the overlay
+   */
+  overlayColor?: string;
+  /**
+   * Render an overlay with custom content
+   */
+  customOverlayContent?: JSX.Element;
+};
+
+type Props = ImageProps & ForwardRefInjectedProps;
```

Comment:
Since you spread `others` in the render block shouldn't your Props include `ViewProps`. and if so export should change to `export default asBaseComponent<Props>(forwardRef(Image));`